### PR TITLE
Upgrade to JUnit 5

### PR DESCRIPTION
**Pull request by juliensadaoui**
Fix #9498

- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our CONTRIBUTING.md document are followed

@hdurix I've upgraded to JUnit 5 in [JHipster server-side library](https://github.com/jhipster/jhipster).

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
impressive, thanks @juliensadaoui !

**Comment by hdurix** [comment]
Great! Thanks :)

File: jhipster-framework/src/test/java/io/github/jhipster/web/filter/reactive/CachingHttpHeadersFilterTest.java [hunk 1]
```diff
@@ -43,8 +41,8 @@ public void cacheHeadersSetWhenPathMatches() {
         WebFilterChain filterChain = (filterExchange) -> {
             try {
                 HttpHeaders headers = filterExchange.getResponse().getHeaders();
-                assertEquals("cache", headers.getPragma());
-                assertEquals("max-age=172800000, public", headers.getCacheControl());
+                assertThat(headers.getPragma()).isEqualTo("cache");
```
Comment (murdos) [inline jhipster-framework/src/test/java/io/github/jhipster/web/filter/reactive/CachingHttpHeadersFilterTest.java]: +1 More assertj!! :)

Comment (murdos) [reply jhipster-framework/src/test/java/io/github/jhipster/web/filter/reactive/CachingHttpHeadersFilterTest.java]: FYI I've also started converting all assertions in code generated by generator-jhipster to assertj (we have a mix of different assertions).

**Comment by DanielFran** [comment]
@juliensadaoui Great work, thanks!

**Pull request merged**